core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -297,6 +297,17 @@ class SnappySessionState(snappySession: SnappySession)
 
   def getTablePartitions(region: CacheDistributionAdvisee): Array[Partition] =
     StoreUtils.getPartitionsReplicatedTable(snappySession, region)
+
+  var isPreparePhase: Boolean = false
```

Comment:
Is there a need of separate variable that too var ? I suppose it can be a method returning "pvs != null" to indicate perpare phase. Moreover, consider using Option[ParamterValueSet] instead of var.